[5.8] Email validation will reject non-string values

paul-thebaud opened the pull request:
In Laravel 5.7, giving a non-string value to a validator was returning a validation error, but in Laravel 5.8, it throw an ErrorException with "Array to string conversion".

This pull-request fix this by checking that the value to validate is a string (using is_string method) or an object which has the __toString method, before validating it using Egulias EmailValidator.

paul-thebaud committed:
Email validation will reject non-string values

paul-thebaud committed:
Fix CS in ValidationValidatorTest

derekmd commented on src/Illuminate/Validation/Concerns/ValidatesAttributes.php:
```diff
@@ -626,7 +626,7 @@ protected function extractDistinctValues($attribute)
     public function validateEmail($attribute, $value)
     {
-        return (new EmailValidator)->isValid($value, new RFCValidation);
+        return is_string($value) && (new EmailValidator)->isValid($value, new RFCValidation);
```
`RFCValidation::isValid()` also accepts an object implementing `__toString()`:

https://github.com/egulias/EmailValidator/blob/master/EmailValidator/Validation/RFCValidation.php#L30

paul-thebaud replied:
I change this. Now, the validation method will first check to have a string or an object implementing __toString, before validating it using RFC.

paul-thebaud commented:
Ok, thank you for the information.
I will update my pull-request to comply with this soon.

paul-thebaud committed:
Email validation accept string or object with __toString method

This commit will resolve the problem pointed by : https://github.com/laravel/framework/pull/27735#discussion_r261703511

paul-thebaud committed:
Fix CS

taylorotwell committed:
Update ValidatesAttributes.php
